File: backend/api_company/serializers/project_task_event.py
from rest_framework import serializers
from api_company.models import ProjectTaskEvent
from api_company.serializers import ProjectHistoryOriginSerializer, ProjectTaskHistoryOriginSerializer


class ProjectTaskEventSerializer(serializers.ModelSerializer):

    class Meta:
        model = ProjectTaskEvent
        fields = ('id', 'comment', 'user_id', 'first_name', 'last_name', 'type', 'project', 'task', 'after_project', 'after_task', 'before_project',
                  'before_task', 'description', 'created_at')


class ProjectTaskEventViewSerializer(serializers.ModelSerializer):

    after_project = ProjectHistoryOriginSerializer(many=False)
    after_task = ProjectTaskHistoryOriginSerializer(many=False)
    before_project = ProjectHistoryOriginSerializer(many=False)
    before_task = ProjectTaskHistoryOriginSerializer(many=False)

    class Meta:
        model = ProjectTaskEvent
        fields = ('id', 'comment', 'user_id', 'first_name', 'last_name', 'type', 'project', 'task', 'after_project', 'after_task', 'before_project',
                  'before_task', 'description', 'created_at')

File: frontend/src/app/models/project-task-event.ts
import {Project} from "./project";
import {ProjectTask} from "./project-task";
import {User} from "./user";

export class ProjectTaskEvent {
    comment?: string;
    // tslint:disable-next-line:variable-name
    user_id?: number;
    // tslint:disable-next-line:variable-name
    first_name?: string;
    // tslint:disable-next-line:variable-name
    last_name?: string;
    user?: User;
    type?: string;
    project?: Project;
    task?: ProjectTask;
    // tslint:disable-next-line:variable-name
    after_project?: Project;
    // tslint:disable-next-line:variable-name
    after_task?: ProjectTask;
    // tslint:disable-next-line:variable-name
    before_project?: Project;
    // tslint:disable-next-line:variable-name
    before_task?: ProjectTask;
    description?: string;
    // tslint:disable-next-line:variable-name
    created_at?: any;
}
